Library/Homebrew/dev-cmd/edit.rb

```diff
@@ -7,6 +7,32 @@
 module Homebrew
   module_function
 
+  module Refinements
+    refine Pathname do
+      sig { returns(T::Boolean) }
+      def core_formula_path?
+        fnmatch?("**/homebrew-core/Formula/**.rb", File::FNM_DOTMATCH)
+      end
+
+      sig { returns(T::Boolean) }
+      def core_cask_path?
+        fnmatch?("**/homebrew-cask/Casks/**.rb", File::FNM_DOTMATCH)
+      end
+
+      sig { returns(T::Boolean) }
+      def core_formula_tap?
+        self == CoreTap.instance.path
+      end
+
+      sig { returns(T::Boolean) }
+      def core_cask_tap?
+        self == CoreCaskTap.instance.path
+      end
+    end
+  end
+
+  using Refinements
```

Comment:
Can this be done inside `def edit` instead so it's not refining when `require`d?

Reply:
Not sure how Ruby works. Do you mean the refinement becomes global at the `module Homebrew` level after the file is read?

Reply:
Anyway, that's not possible.
```
Error: Module#using is not permitted in methods
````